### DISCUSSION TITLE
Issues on external devices?

### DISCUSSION THREAD

**Discussion by oliver-hekmatara** [gilbarbara/react-spotify-web-playback#178] [General]
Hi

The player works great when I play tracks through the web player but whenever I switch to external speakers I'm getting into trouble.

I can play tracks but controlling the playback, switching tracks and so on doesn't really work. On prev/next it just play/pause the currently playing track. Have you had any issues with this @gilbarbara? This is basically the only thing left for me to solve, but it's a crucial feature for my app.

**Comment by gilbarbara** [comment]
Hey @oliver-hekmatara 
What do you mean by External speakers? Is it another Spotify device or just an external speaker connected to your computer?
I tried both scenarios using the demo, and they worked as expected. Can you try with the demo?
If you can reproduce it, send a link to a stackblitz or codesandbox example.

**Comment by oliver-hekmatara** [comment]
Yes, exactly, I mean a device that is not the custom web player. Switching tracks with prev/next just quickly pause & play the track. This issue could of course be on my side but since everything works fine with the "internal" web player I thought it could be something with external device playback on your side. I will do some more testing. 

I can't try with the demo, unfortunately its not loading/working for me at all. It loads the player but there's no track loaded and I don't see any device selector.

**Reply by gilbarbara** [reply]
The demo with this [URL](https://react-spotify-web-playback.gilbarbara.dev/)? 
Try to log out and log in again.
Also, check the network panel since Stopify's API handles all the external player requests.

**Reply by oliver-hekmatara** [reply]
Yes, that's the one I'm trying. Getting an error in the console "Uncaught (in promise) SyntaxError: Unexpected token 'C', "Check sett"... is not valid JSON".

But I've found the reason why it's not working with external device (I think). I'm controlling the playback both from your player but also from a custom playlist I've built. For some reason I didnt get it to work very well with a list of track URI:s and offset, so my solution was to only have one single track URI active at all time. So basically when switching track I'm just switching out the track URI. But this way of controlling the playback don't seem to work with external devices, only with the custom web player. So I'm looking into how I could manage this with a list of track URI:s and offset instead. (so far it seems very challenging to create a smooth experience like this, keeping the player and playlist in sync)

**Reply by gilbarbara** [reply]
Weird, I'll check that.

Yeah, it can be tricky, but it goes beyond the scope of this library. The callback will help with the sync anyway.
That's how I'm doing it with my app to generate playlists. You can check it out at https://beta.musicbot.io.
